Corona-Warn-App/src/main/java/de/rki/coronawarnapp/task/TaskFactory.kt

```diff
@@ -1,11 +1,18 @@
 package de.rki.coronawarnapp.task
 
+import org.joda.time.Duration
+
 interface TaskFactory<
     ProgressType : Task.Progress,
     ResultType : Task.Result
     > {
 
     interface Config {
+        /**
+         * The maximal runtime of the task, before it is canceled by the controller
+         */
+        val executionTimeout: Duration
```

Comment:
I'm wondering if made  sense  to 
a) set a default
b) enforce a maximum timeout

Reply:
> set a default
Because we (the TaskController itself) doesn't know how the tasks will be used, I don't think we can set a reasonable timeout, and I like that when someone creates a new task, they are forced to think about and specify a timeout.

> enforce a maximum timeout
The maximum timeout is basically Long.MAX_VALUE in milliseconds. Similar to my previous point, the TaskController should not be aware or take part in specifying too much default behavior for Tasks.